tests/test_siamese_dataset.py

```diff
@@ -0,0 +1,126 @@
+import sys
+sys.path += ["."]  # Python 3 hack
+import numpy as np
+import h5py
+import os
+import Util
+from SiameseDataset import SiameseHDFDataset
+import sys
+import unittest
+import better_exchook
+better_exchook.install()
+better_exchook.replace_traceback_format_tb()
+Util.initThreadJoinHack()
+
+
+def generate_dummy_hdf(num_datasets=1):
+  for idx in range(1, num_datasets + 1):
+    dataset = h5py.File('./dummy.%i.hdf5' % idx, 'w')
+    dataset.create_group('streams')
+
+    dataset['streams'].create_group('features')
+    dataset['streams']['features'].attrs['parser'] = "feature_sequence"
+    dataset['streams']['features'].create_group('data')
+
+    dataset['streams'].create_group('classes')
+    dataset['streams']['classes'].attrs['parser'] = "sparse"
+    dataset['streams']['classes'].create_group('data')
+
+    import string
+    random = np.random.RandomState()
+    feature_size = 13
+    seq_names = ['dataset_%d_sequence_%d' % (idx, i) for i in range(100)]
+    class_names = list(string.ascii_lowercase)
+    num_classes = len(class_names)
+    print(class_names, len(class_names))
+    for idx in range(100):
+      class_id = random.randint(low=0, high=num_classes)
+      seq_len  = random.randint(low=1, high=20)
+      features = random.rand(seq_len, feature_size)
+      dataset['streams']['features']['data'].create_dataset(name=seq_names[idx], data=features, dtype='float32')
+      dataset['streams']['classes']['data'].create_dataset(name=seq_names[idx], shape=(1,), data=np.int32(class_id), dtype='int32')
+
+    dt = h5py.special_dtype(vlen=str)
+    feature_names = dataset['streams']['classes'].create_dataset("feature_names", shape=(len(class_names),), dtype=dt)
+    for id_x, orth in enumerate(class_names):
+      feature_names[id_x] = orth
+
+    dt = h5py.special_dtype(vlen=str)
+    sequence_names_data = dataset.create_dataset("seq_names", shape=(len(seq_names),), dtype=dt)
+    for ind, val in enumerate(seq_names):
+      sequence_names_data[ind] = val
+
+    dataset.close()
+  return ['./dummy.%i.hdf5' % idx for idx in range(1, num_datasets + 1)]
+
+
+def test_triplet_sampling():
```

Comment:
Do not add this to a new separate test file. This can be part of `test_HDFDataset.py`.